### PR TITLE
Migrate from fs-extra-promise back to fs-extra

### PR DESCRIPTION
**Pull request by KendrickAng**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Other, please explain:
This PR maintains existing features while removing dependency fs-extra-promise in favor of fs-extra, which is more regularly updated and maintained.

Address #1194 .

**What is the rationale for this request?**
As posted originally by Zeyu:
fs-extra-promise originally provided promisified versions of fs-extra's methods.
It hasn't been updated in quite some time however, and documentation on it is more difficult to find than fs-extra.
fs-extra has also been updated to provide promisified versions quite a while ago, and is still being actively maintained, with widely available documentation.

**What changes did you make? (Give an overview)**
Replaces all fs-extra-promise methods with their equivalent fs-extra methods.

Here is a brief document comparing the changes between `fs-extra` and `fs-extra-promise`:
https://docs.google.com/document/d/1-HCLbtx7GslshHjaZI9bIbVW79wGvB_UPWJZThgFLYI/edit?usp=sharing

In short, most difference in functionality would arise from `fs-extra-promise` using older versions of `fs-extra`, an example being that JSON in older versions was being written to files beautified automatically; this has to be specified in the current `fs-extra`.

**Provide some example code that this change will affect:**

```js

```

**Is there anything you'd like reviewers to focus on?**
There are some relatively non-trivial changes in core\src\Page\index.js line 1116 and core\src\Page\index.js line 1111, please have a look at them and verify those changes are acceptable.

**Testing instructions:**
Serve the docs folder and verify that functionality has not changed.

**Proposed commit message: (wrap lines at 72 characters)**
Migrate from fs-extra-promise back to fs-extra

### CODE REVIEW

File: packages/core/src/Site/index.js [hunk 1]
```diff
@@ -1269,7 +1270,8 @@ class Site {
           })),
       };
 
-      fs.outputJsonAsync(siteDataPath, siteData)
+      // TODO: figure out why the only usage of fse breaks when changed to outputJson.
+      fse.outputJsonAsync(siteDataPath, siteData)
```
Comment (KendrickAng) [inline packages/core/src/Site/index.js:1274]: @ang-zeyu I'm having some trouble at this part; changing this line to `fs.outputJson(siteDataPath, siteData)` causes `test_site/_site/siteData.json` to output the correct json, but incorrectly formatted. For example"
image

On the other hand, leaving this statement as it is outputs a correctly formatted json file:
image

Do you happen to know why this is so? I suspected usage of a JSON.stringify(.., null, 2) but couldn't find any. Or maybe it's some side-effect of another library?

Comment (ang-zeyu) [reply packages/core/src/Site/index.js:1274]: I'm assuming the tests fail because `outputJson` is not beautified
Try the options here https://github.com/jprichardson/node-fs-extra/blob/master/docs/outputJson.md

for 3rd party libraries, always go back to the documentation to double check the behaviour first if there are issues =P
For this pr especially, do check if there are obscure differences between the method pairs (e.g. `copy` vs `copyAsync`)

Comment (KendrickAng) [reply packages/core/src/Site/index.js:1274]: noted! I'll be sure to note down the differences, if any

File: packages/core/src/Site/index.js [hunk 2]
```diff
@@ -1269,7 +1270,9 @@ class Site {
           })),
       };
 
-      fs.outputJsonAsync(siteDataPath, siteData)
+      fse.outputJsonAsync();
```
Comment (ang-zeyu) [inline packages/core/src/Site/index.js:1273]: 👀 

let's remove all of it:
- there's a few redundant mocks for fs-extra-promise still as well;
- the walk-sync mock which relies on it is also redundant
- package.json entries, then updating `package-lock`

Comment (KendrickAng) [reply packages/core/src/Site/index.js:1273]: 😓

**Comment by KendrickAng** [comment]
I've added the following changes:
* removed the fs-extra-promise mock (since no tests rely on fs-extra-promise anymore)
* removed fs-extra-promise from package.json, regenerated package-lock.json with `npm run setup`
* changed walk-sync to rely on fs-extra itself rather than the fs-extra-promise mock

File: packages/core/src/Page/index.js [hunk 3]
```diff
@@ -1110,11 +1110,10 @@ class Page {
       const srcPath = path.resolve(plugin._pluginAbsolutePath, assetPath);
       const srcBaseName = path.basename(srcPath);
 
-      fs.existsAsync(plugin._pluginAssetOutputPath)
-        .then(exists => exists || fs.mkdirp(plugin._pluginAssetOutputPath))
+      fs.ensureDir(plugin._pluginAssetOutputPath)
```
Comment (ang-zeyu) [inline packages/core/src/Page/index.js:1113]: 👍

**Review by ang-zeyu — APPROVED** [review]
> Here is a brief document comparing the changes between fs-extra and fs-extra-promise:
> https://docs.google.com/document/d/1-HCLbtx7GslshHjaZI9bIbVW79wGvB_UPWJZThgFLYI/edit?usp=sharing

Thanks for the thorough comparison!

This looks good 👍

---

Regarding the walk-sync mock, did removing it (and the `jest.mock` call) not work?
Since we have a `jest.mock('fs')` call in the same file which _should_ propagate to walk-sync

But it can be investigated (if it was even necessary when added) and dealt with at a later time 

---

> Address #1194

Small nit: you can use one of these when the pr completely resolves the issue 🙂
https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords
